### PR TITLE
`pallet_gear`: add reset extrinsic

### PR DESCRIPTION
**Pull request by ukint-vs**
- reset storage prefixes defined in `gear_common`
- reset storage values defined in `pallet_gear`

todo:
- [x] update runtime-upgrade workflow

@gear-tech/dev

### CODE REVIEW

File: common/src/lib.rs [hunk 1]
```diff
@@ -419,6 +419,19 @@ pub fn code_exists(code_hash: H256) -> bool {
...
+    sp_io::storage::clear_prefix(STORAGE_PROGRAM_PREFIX, None);
+    sp_io::storage::clear_prefix(STORAGE_PROGRAM_PAGES_PREFIX, None);
+    sp_io::storage::clear_prefix(STORAGE_MESSAGE_PREFIX, None);
+    sp_io::storage::clear_prefix(STORAGE_MESSAGE_NONCE_KEY, None);
+    sp_io::storage::clear_prefix(STORAGE_MESSAGE_USER_NONCE_KEY, None);
+    sp_io::storage::clear_prefix(STORAGE_CODE_PREFIX, None);
+    sp_io::storage::clear_prefix(STORAGE_CODE_METADATA_PREFIX, None);
+    sp_io::storage::clear_prefix(STORAGE_CODE_REFS_PREFIX, None);
+    sp_io::storage::clear_prefix(STORAGE_WAITLIST_PREFIX, None);
+    sp_io::storage::clear_prefix(GAS_VALUE_PREFIX, None);
```
Comment (shamilsan) [inline common/src/lib.rs:432]: Code prefix includes metadata and refs, and message prefix includes nonce.

```suggestion
    sp_io::storage::clear_prefix(STORAGE_PROGRAM_PREFIX, None);
    sp_io::storage::clear_prefix(STORAGE_PROGRAM_PAGES_PREFIX, None);
    sp_io::storage::clear_prefix(STORAGE_MESSAGE_PREFIX, None);
    sp_io::storage::clear_prefix(STORAGE_CODE_PREFIX, None);
    sp_io::storage::clear_prefix(STORAGE_WAITLIST_PREFIX, None);
    sp_io::storage::clear_prefix(GAS_VALUE_PREFIX, None);
```

Alternatively we can clear `g::` prefix at once but it is quite dangerous.

**Comment by ekovalev** [comment]
Stupid question: why do we need this at all?

**Comment by ukint-vs** [comment]
> Stupid question: why do we need this at all?

this will allow us to update the runtime without migrations and resetting the entire network

File: common/src/lib.rs (hunk 1 above)
Comment (ukint-vs) [reply common/src/lib.rs:432]: anything can be stored under `g::` prefix, so this is definitely not suitable

**Review by NikVolf — APPROVED** [review]

**Pull request merged**